### PR TITLE
feat: add partition metrics table

### PR DESCRIPTION
**Pull request by zoercai**
### TODO List:

- [x] Refactor, find nits etc
- [x] Add TTL

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/DetectNewPartitionsDoFn.java [hunk 1]
```diff
@@ -175,6 +178,7 @@ private PartitionMetadata updateToScheduled(PartitionMetadata partition) {
     final com.google.cloud.Timestamp scheduledAt = partitionMetadataDao.updateToScheduled(token);
 
     if (InitialPartition.isInitialPartition(token)) {
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/DetectNewPartitionsDoFn.java:180]: I think that if you `pull spanner-cdc` you shouldn't see this branching path anymore.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PipelineInitializer.java [hunk 2]
```diff
@@ -53,5 +60,6 @@ private static void createFakeParentPartition(
             .setState(State.CREATED)
             .build();
     partitionMetadataDao.insert(parentPartition);
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PipelineInitializer.java:62]: Could we do both these operations in the same transaction? I would vote for having just one `Dao` for both tables, but let me know what you think.

Comment (thiagotnunes) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PipelineInitializer.java:62]: The same for the other operations, if we can update the state of both the metadata table and the metrics table in the same transaction that would be best. We could also think about removing some of the state from the metadata table and leave it only in the metrics table.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/dao/PartitionMetricsAdminDao.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            + " TIMESTAMP OPTIONS (allow_commit_timestamp=true),"
+            + COLUMN_LAST_UPDATED_AT
+            + " TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true),"
+            + ") PRIMARY KEY (PartitionToken)";
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op =
+        databaseAdminClient.updateDatabaseDdl(
+            instanceId, databaseId, Collections.singletonList(metricsCreateStmt), null);
+    try {
+      // Initiate the request which returns an OperationFuture.
+      op.get(TIMEOUT_MINUTES, TimeUnit.MINUTES);
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/dao/PartitionMetricsAdminDao.java:93]: Could we log the name of the metrics table in the SpannerIO class?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PipelineInitializer.java (hunk 2 above)
Comment (zoercai) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/PipelineInitializer.java]: I've merged the metrics DAO operations into the metadata DAO. For the metadata columns and duplicate metrics I will create another PR.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/actions/QueryChangeStreamAction.java [hunk 4]
```diff
@@ -102,6 +104,7 @@ public QueryChangeStreamAction(
               partition.isInclusiveEnd(),
               partition.getHeartbeatMillis())) {
 
+        int recordsProcessed = 0;
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/actions/QueryChangeStreamAction.java:107]: nit: Could we make this as a long? Just to be on the safe side if there are long lived partitions / high throughput.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/actions/QueryChangeStreamAction.java [hunk 5]
```diff
@@ -117,21 +120,26 @@ public QueryChangeStreamAction(
               maybeContinuation =
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/actions/QueryChangeStreamAction.java:120]: nit: maybe move the `recordsProcessed++` as the first statement within the `for` loop?

**Review by thiagotnunes — APPROVED** [review]

**Pull request merged**